### PR TITLE
Do not fail a sandboxed spawn for missing execution statistics

### LINKED ISSUE #22151: failed: I/O exception during sandboxed execution: No such file or directory
### Description of the bug:

Periodically, the following error would occur when running tests

```
 Testing <blah> failed: I/O exception during sandboxed execution: /dev/shm/bazel-sandbox.34e9fe25bb0c6624a8ba8f5a00a18c3243dfc943119e415e09934c77b955441f/linux-sandbox/148/stats.out (No such file or directory)
 ```
 
 we're on bazel 6.5.0 with spawn strategy `linux-sandbox`, jobs set to 1:1 with vcpus, sandbox mounted at `/dev/shm` 
 
 Whenever this occurs, we see system memory usage at 82-83% with cpu maxed at 100%. 
 

### Which category does this issue belong to?

_No response_

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

 I don't have a reliable repro. it happens sporadically

### Which operating system are you running Bazel on?

ubuntu 20.04

### What is the output of `bazel info release`?

release 6.5.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

_No response_

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?

_No response_

### Is this a regression? If yes, please try to identify the Bazel commit where the bug was introduced.

this has been occuring more frequently since we switched to 6.5.0 from 6.3.2

### Have you found anything relevant by searching the web?

_No response_

### Any other information, logs, or outputs that you want to share?

_No response_

### PR DESCRIPTION
**Pull request by fmeum**
This was already the case for "local" spawns. Statistics may be missing if the spawn wrapper exits abnormally.

Fixes #22151

### CODE REVIEW

**Comment by fmeum** [comment]
@bazel-io fork 7.3.0

**Comment by fmeum** [comment]
@bazel-io fork 7.2.1

**Comment by fmeum** [comment]
I marked this for cherry-picking as users reported on Slack that this is blocking their Bazel 7 update and the fix is very safe.

File: src/main/java/com/google/devtools/build/lib/shell/ExecutionStatistics.java [hunk 1]
```diff
@@ -31,6 +31,11 @@ public final class ExecutionStatistics {
    */
   public static Optional<ResourceUsage> getResourceUsage(Path executionStatisticsProtoPath)
       throws IOException {
+    if (!executionStatisticsProtoPath.exists()) {
+      // Collecting resource usage is best-effort and the file may be missing if the wrapper around
+      // the command terminated abnormally.
+      return Optional.empty();
+    }
```
Comment (tjgq) [inline src/main/java/com/google/devtools/build/lib/shell/ExecutionStatistics.java:38]: Nit: catch a FileNotFoundException below instead.

**Review by tjgq — CHANGES_REQUESTED** [review]

**Review by tjgq — APPROVED** [review]
Thanks!

**Comment by fmeum** [comment]
@tjgq Added a "Fixes" line to the PR description.

**Pull request closed without merging**